qiskit/ignis/verification/randomized_benchmarking/fitters.py

```diff
@@ -750,3 +763,189 @@ def plot_rb_data(self, pattern_index=0, ax=None,
+        Args:
+            new_purity_result: list of purity rb results
+            rerun_fit: re-caculate the means and fit the result
+            Additional information:
+            Assumes that 'result' was executed is
+            the output of circuits generated by randomized_becnhmarking_seq
+            where is_purity = True.
+        """
+        for d in range(self._npurity):
+            self.rbfit_pur[d].add_data(new_purity_result[d], rerun_fit)
+
+        if rerun_fit:
+            self.calc_data()
+            self.calc_statistics()
+            self.fit_data()
+
+    def calc_data(self):
+        """
+        Retrieve probabilities of success from execution results.
+        Outputs results into an internal variables _raw_data.
+        Calculates the sum of the raw_data from the 3^n
+        correlators.
+        """
+        for d in range(self._npurity):
+            self.rbfit_pur[d].calc_data()
+
+        # Update the raw_data of the sum of correlators
+        self.rbfit_pur[self._npurity].raw_data = [
+            [[] for _ in range(len(self.seeds[0]))]
+            for _ in range(len(self._rb_pattern))]
+
+        for patt_ind in range(len(self._rb_pattern)):
+            for seed_ind in range(len(self.seeds[0])):
+                qubits = self._rb_pattern[patt_ind]
+                new_raw_data = np.zeros(len(self._cliff_lengths[patt_ind]))
+                for d in range(self._npurity):
+                    new_raw_data += self.raw_data_pur[d][patt_ind][seed_ind]
+                new_raw_data = new_raw_data / (2 ** len(qubits))
+                self.rbfit_pur[self._npurity].raw_data[patt_ind][seed_ind] = \
+                    new_raw_data
+
+    def calc_statistics(self):
+        """
+         Extract averages and std dev.
+         Calculates the average and std of the
+         sum of the 3^n correlators.
+         """
+        for d in range(self._npurity):
+            self.rbfit_pur[d].calc_statistics()
+        self.rbfit_pur[self._npurity].calc_statistics()
+
+    def fit_data_pattern(self, patt_ind, fit_guess):
+        """
+        Fit the sum of the results of all correlators
+        of a particular pattern, to an exponential curve.
+        Args:
+            patt_ind: index of the data to fit
+            fit_guess: guess values for the fit
+        """
+        self.rbfit_pur[self._npurity].fit_data_pattern(
+            patt_ind, fit_guess)
+
+    def fit_data(self):
+        """
+         Fit the sum of the results of all correlators
+         to an exponential curve. Fit each of the patterns.
+         Puts the results into a list of fit dictionaries:
+         where each dictionary corresponds to a pattern and has fields:
+             'params' - three parameters of rb_fit_fun. The middle one is the
+                        exponent.
+             'err' - the error limits of the parameters.
+             'epc' - error per Clifford
+         """
+        self.rbfit_pur[self._npurity].fit_data()
```

Comment:
The purity EPC is not the same as the standard RB EPC.

Reply:
alpha for purity RB should be defined using the same function as in standard RB: 
A+B*p^m (alpha=p)
So, how the EPC for purity RB is defined?

Reply:
I will fix it.